src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp

```diff
@@ -690,6 +690,61 @@ TEST_F(TestEnlsvgPathPlanner,
     TestUtil::checkPathDoesNotIntersectObstacle(path_points, obstacles);
 }
 
+TEST_F(TestEnlsvgPathPlanner, test_going_around_defense_area)
+{
+    Field field              = Field::createSSLDivisionAField();
+    Rectangle navigable_area = field.fieldBoundary();
+
+    Point start{4.13, -1.97}, dest{2.02, 1.96};
+
+    std::vector<Polygon> obstacle_polygons{
+        Rectangle(Point(3.018497, -1.481503), Point(4.8, 1.481503)),
+    };
+
+    std::vector<ObstaclePtr> obstacles = {
+        robot_navigation_obstacle_factory.createFromShape(
+            Rectangle(Point(3.018497, -1.481503), Point(4.8, 1.481503))),
+    };
+
+    EnlsvgPathPlanner planner =
+        EnlsvgPathPlanner(navigable_area, obstacles, field.boundaryMargin());
+    std::optional<Path> path = planner.findPath(start, dest);
+
+    ASSERT_TRUE(path != std::nullopt);
+
+    std::vector<Point> path_points = path->getKnots();
+
+    // Make sure the start and end points match
+    EXPECT_EQ(start, path->getStartPoint());
+    EXPECT_EQ(dest, path->getEndPoint());
+
+    // Make sure path does not exceed a bounding box
+    Rectangle bounding_box({1, -2.5}, {5, 5});
+    TestUtil::checkPathDoesNotExceedBoundingBox(path_points, bounding_box);
+    TestUtil::checkPathDoesNotIntersectObstacle(path->getKnots(), obstacle_polygons);
+
+    // Make sure that reusing the path planner still works by computing another path
+    start = Point(5, 0);
+    dest  = Point(-2, 0.5);
+
+    auto path_two = planner.findPath(start, dest);
+    path_points   = path_two->getKnots();
+    ASSERT_TRUE(path_two != std::nullopt);
+
+    path_points = path_two->getKnots();
+
+    // Make sure the start and end points match
+    EXPECT_EQ(start, path_two->getStartPoint());
+    EXPECT_EQ(dest, path_two->getEndPoint());
+
+    EXPECT_LE(4, path_points.size());
+
+    // Make sure path does not exceed a bounding box;
+    bounding_box = Rectangle(Point(5.1, 1.8), Point(-2.1, -0.1));
+    TestUtil::checkPathDoesNotExceedBoundingBox(path_points, bounding_box);
+    TestUtil::checkPathDoesNotIntersectObstacle(path_points, obstacles);
```

Comment:
this part isn't actually necessary, bad copy pasta on my part

Reply:
fixed